megatron/model/transformer.py

```diff
@@ -448,24 +448,21 @@ def __init__(self, init_method, output_layer_init_method,
             self.mlp = ParallelMLP(init_method,
                                output_layer_init_method)
         else:
-            if not args.ds_inference or self.num_experts > dist.get_world_size():
-                moe_mp_size = 1
-            else:
-                moe_mp_size = dist.get_world_size() // self.num_experts
-            
+            enable_expert_tensor_parallelism = args.enable_expert_tensor_parallelism
             self.mlp = MoE(args.hidden_size,
                             ParallelMLP(init_method,
                                 output_layer_init_method=output_layer_init_method,
-                                MOE=True,
-                                MoE_mp_size=moe_mp_size),
+                                moe=True,
+                                enable_expert_tensor_parallelism=enable_expert_tensor_parallelism),
                             num_experts=self.num_experts, 
-                            ep_size=args.moe_expert_parallel_size,
+                            ep_size=ep_size,
```

Comment:
This actually leads to an error since ep_size is not defined. After rolling back one commit (8398bb05b62249909117f069683da267a87cf960) the error disappeared and non-expert/expert MP work fine. Please make sure to fix this before merging. And still need full code review from @awan-10 .

Reply:
fixed in the next commit